### DISCUSSION TITLE
Trancuated data in DBEAVER

### DISCUSSION THREAD

**Discussion by Jivelannag** [dbeaver/dbeaver#30027] [Q&A]
Hello!
I am using a free version of DBEAVER for SQL practise. 
I have stumbled upon a problem with a converting function. 
And I think that the problem occurred because of 'comma' that always shows up when an integer or a date type value reaches to 1000. 
for example I have a column "released year" so instead of storing 2023. It stores 2,203. 
MySQL does not convert columns into another datatype, it constantly gives me an error 1265
<img width="212" alt="Screenshot 2024-05-08 at 22 18 05" src="https://github.com/dbeaver/dbeaver/assets/165540354/31322652-895c-494d-9631-07395dbd002f">
Sometimes DBEAVER converts columns that contain numerical value into text/varchar value even if there is no difference between columns in terms of length of input values. For example, I have a column that was imported as varchar with very small numbers while other column was imported as Integer with bigger numbers. 
<img width="1041" alt="Screenshot 2024-05-08 at 22 25 38" src="https://github.com/dbeaver/dbeaver/assets/165540354/6563f176-dad6-439f-9ee0-6450bf49defc">
I have tried to convert varchar datatype into INTEGER, DOUBLE, REAL. Always error "data trancured"
Even when I re-import data into my database. 
Could, please, someone, help me? 
I will highly appreciate it. 
Thank you.

**Comment by ghost** [comment]
Привет Аня. Это может быть вызвано различиями в формате и типе данных между исходным и целевым столбцом. Перед попыткой преобразования столбца убедись, что данные соответствуют формату целевого типа данных.
   Если ты пыталась изменить тип данных столбца из VARCHAR в INTEGER, DOUBLE или REAL, но получаете ошибку "данные перегружены", это может быть связано с тем, что некоторые значения в столбце не могут быть корректно преобразованы в числовой формат.
   Убедитесь, что все значения в столбце корректно представлены в числовом формате.
Попробуй очистить столбец от любых специальных символов или лишних пробелов перед попыткой преобразования.
Попробуй создать новый столбец с нужным типом данных и скопировать данные из старого столбца в новый, используя функцию преобразования, например, CAST или CONVERT.
